shared/events.js

```diff
@@ -88,6 +92,14 @@ export class HashableEntry extends Hashable {
     root.add(msg)
     return msg
   }
+  static isHashableEntry (obj) {
+    if (!obj) return false
+    if (typeof obj.version !== 'number') return false
+    if (typeof obj.parentHash !== 'string') return false
+    if (!obj.data) return false
+    if (typeof obj.type !== 'string') return false
+    return true
+  }
```

Comment:
This shouldn't exist either. The correct way is to index the exported classes and attempt to deserialize via `fromObject`. **Only** upon successful return from that function is it considered to be valid.

See line 272 in state.js for how to do this, i.e.:

```js
(Events[type] || contracts[type]).fromObject(entry, hash)
```